x/auth/types/account_retriever.go

```diff
@@ -1,67 +1,65 @@
 package types
 
 import (
-	"fmt"
+	context "context"
 
 	"github.com/cosmos/cosmos-sdk/client"
-	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
```

Comment:
Is this similar to what you had in mind @aaronc?

Reply:
This totally makes sense. I was just talking about a gRPC service that calls this `AccountRetriever` under the hood and let's it take care of all the `AccountI` stuff.